build-native.sh

```diff
@@ -116,25 +91,53 @@ pushd "$OBJECT_DIR" > /dev/null 2>&1
 # let's disable set -e so we can popd out of the object directory if they fail.
 set +e
 
-echo "CC=$CC"
-echo "CXX=$CXX"
-
-(set -x ; cmake -DCMAKE_BUILD_TYPE="$CMAKE_BUILD_TYPE" -DCMAKE_INSTALL_PREFIX="$BINARY_DIR" -G "Unix Makefiles" "$SOURCE_ROOT")
-
+#Call libuv build commands
+echo "Commencing build of native compenents"
+echo .
+echo $SUBMODULE_ROOT
+pushd $SUBMODULE_ROOT
+echo "Building with autotools"
+(set -x; sh autogen.sh)
 if [ $? -ne 0 ]; then
-    echo "ERROR: Could not generate build scripts, cmake failed."
+    echo "ERROR: Build failed."
     popd > /dev/null 2>&1
     exit 1
 fi
-
-(set -x ; make install -j)
-
+(sh configure --prefix=$BINARY_DIR)
+(set x; make -j)
 if [ $? -ne 0 ]; then
     echo "ERROR: Build failed."
     popd > /dev/null 2>&1
     exit 1
 fi
-
+(set x; make check -j)
```

Comment:
I think this target will run tests, correct?  Could we make that optional or just remove it for now and add it back if we need it?